src/crud.py

```
@@ -35,6 +36,92 @@ async def embed(self, query: str) -> list[float]:
+        # Note: Schema limits ensure we'll never exceed the 2,048 inputs per request limit
+        MAX_TOKENS_PER_REQUEST = 300_000
+        MAX_TOKENS_PER_INPUT = 8192
+
+        for i, (text_id, text) in enumerate(zip(all_ids, all_texts, strict=False)):
+            # Get pre-computed token count
+            text_tokens = all_token_counts[i]
+
+            # Skip if single text exceeds per-input limit. This is a safety measure but realistically shouldn't really happen
+            if text_tokens > MAX_TOKENS_PER_INPUT:
+                logger.warning(
+                    f"Skipping embedding for {text_id} - {text_tokens} tokens exceeds limit"
```

Ruff (0.11.9)

Logging statement uses f-string

(G004)